lisa/src/test/java/it/unive/lisa/test/imp/tutorial/Parity.java

```diff
@@ -0,0 +1,220 @@
+package it.unive.lisa.test.imp.tutorial;
+
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.BaseNonRelationalValueDomain;
+import it.unive.lisa.cfg.type.Type;
+import it.unive.lisa.symbolic.value.BinaryOperator;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.Identifier;
+import it.unive.lisa.symbolic.value.TernaryOperator;
+import it.unive.lisa.symbolic.value.UnaryOperator;
+
+public class Parity extends BaseNonRelationalValueDomain<Parity> {
+
+	private enum Values {
+		ODD,
+		EVEN,
+		TOP,
+		BOT
+	}
+
+	private final Values parity;
+
+	private Parity(Values parity) {
+		this.parity = parity;
+	}
+
+	public Parity() {
+		this(Values.TOP);
+	}
+
+	@Override
+	public Parity top() {
+		return new Parity(Values.TOP);
```

Comment:
We can make the top value a singleton, removing also the `isTop` method. Same could be done for the bottom value.